app/models/aimodels/AiModel.scala

```diff
@@ -87,7 +112,32 @@ class AiModelDAO @Inject()(sqlClient: SqlClient)(implicit ec: ExecutionContext)
       )
 
   override protected def readAccessQ(requestingUserId: ObjectId): SqlToken =
-    q"_organization IN (SELECT _organization FROM webknossos.users_ WHERE _id = $requestingUserId)"
+    q"""_id IN (
+          SELECT a._aiModel
+          FROM webknossos.aiModel_organizations AS a
+          INNER JOIN webknossos.organizations AS o
+            ON a._organization = o._id
+          WHERE
+              (o._id IN (
+                  SELECT _organization
+                  FROM webknossos.users_
+                  WHERE _multiUser = (
+                      SELECT _multiUser
+                      FROM webknossos.users_
+                      WHERE _id = $requestingUserId
+                  )
+              ))
+              OR TRUE IN (
+                  SELECT isSuperUser
+                  FROM webknossos.multiUsers_
+                  WHERE _id IN (
+                      SELECT _multiUser
+                      FROM webknossos.users_
+                      WHERE _id = $requestingUserId
+                  )
+              )
+          )
+     """
```

Comment:
I adjusted the accessor query so that all models are shown to the user which the user has potentially access to -> all organization in which the user is part of or allowed to join because of being a super user.

While testing I realized that this might be unwanted as e.g. super users can see all models all the time therefore potentially cluttering the view / list of models.

@fm3 Do you think it makes sense to only show models of the organization in which the user is currently logged in?

Reply:
Yes I agree, only the organization the user is currently logged in to should be considered.